pkg/controllers/nodeclass/controller.go

```diff
@@ -80,6 +95,31 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeCl
 	return reconcile.Result{RequeueAfter: 5 * time.Minute}, err
 }
 
+func (c *Controller) Finalize(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
+	stored := nodeClass.DeepCopy()
+	if !controllerutil.ContainsFinalizer(nodeClass, v1beta1.TerminationFinalizer) {
+		return reconcile.Result{}, nil
+	}
+	nodeClaimList := &corev1beta1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, nodeClaimList, client.MatchingFields{"spec.nodeClass.name": nodeClass.Name}); err != nil {
+		return reconcile.Result{}, fmt.Errorf("listing NodeClaims using NodeClass, %w", err)
+	}
+	if len(nodeClaimList.Items) > 0 {
+		c.recorder.Publish(WaitingOnNodeClaimTerminationEvent(nodeClass, lo.Map(nodeClaimList.Items, func(nc corev1beta1.NodeClaim, _ int) string { return nc.Name })))
+		return reconcile.Result{RequeueAfter: time.Minute * 10}, nil // periodically fire the event
```

Comment:
We probably don't need this 10s anymore, right?

Reply:
I didn't do 10s here, I did 10m. I was thinking about it and thought that a periodic event to keep that event fresh on the NodeClass was important. Otherwise, you'd only get the event once and then it would be dropped by etcd after an hour